platform/posix/transport/src/transport_reconnect_posix.c

```diff
@@ -20,50 +20,60 @@
  */
 
 /**
- * @file reconnect_posix.c
+ * @file transport_reconnect_posix.c
  * @brief Implementation of the backoff logic when connection fails to the server fails.
  */
 
 /* Standard includes. */
 #include <unistd.h>
 #include <stdlib.h>
 #include <time.h>
-#include "reconnect.h"
+#include "transport_reconnect.h"
 
 /*-----------------------------------------------------------*/
 
 bool Transport_ReconnectBackoffAndSleep( TransportReconnectParams_t * pReconnectParams )
 {
     bool status = false;
-    uint32_t jitter = 0;
+    uint32_t nextBackOffAndJitter = 0;
 
     /* If MAX_RECONNECT_ATTEMPTS is set to 0, try forever */
-    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPS ) ||
-        ( 0 == MAX_RECONNECT_ATTEMPS ) )
+    if( ( pReconnectParams->attemptsDone < MAX_RECONNECT_ATTEMPTS ) ||
+        ( 0 == MAX_RECONNECT_ATTEMPTS ) )
     {
+        /* Reduce backoff value if more than max backoff time value. */
+        if( pReconnectParams->nextBackOffSec > MAX_RECONNECT_BACKOFF_SECONDS )
```

Comment:
Is this necessary check ?  pReconnectParams->nextBackOffSec  is always going to be <= MAX_RECONNECT_BACKOFF_SECONDS based on line 67

Reply:
Sure, removed it _again_